codegen crashes if out param variable is created in function call

_From @ghosttie on February 28, 2017 23:46_

If I create an out param variable in the function call:

```
			if (Guid.TryParse(stringValue, out var valueAsGuid)) {
```

I get this exception:

> There was an error running the selected code generator:
> 'There was an error creating a DbContext :C:\Users\jmunro\Dropbox\Work\projects\asp.net\FileVision Application Server\FileVision Application Server\GuidHeaderModelBinder.cs(20,43): error CS1003: Syntax error, ',' expected
> C:\Users\jmunro\Dropbox\Work\projects\asp.net\FileVision Application Server\FileVision Application Server\GuidHeaderModelBinder.cs(20,39): error CS0103: The name 'var' does not exist in the current context
> C:\Users\jmunro\Dropbox\Work\projects\asp.net\FileVision Application Server\FileVision Application Server\GuidHeaderModelBinder.cs(20,43): error CS0103: The name 'valueAsGuid' does not exist in the current context
> C:\Users\jmunro\Dropbox\Work\projects\asp.net\FileVision Application Server\FileVision Application Server\GuidHeaderModelBinder.cs(21,56): error CS0103: The name 'valueAsGuid' does not exist in the current context
>    at Microsoft.VisualStudio.Web.CodeGeneration.ActionInvoker.<BuildCommandLine>b__6_0()
>    at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
>    at Microsoft.VisualStudio.Web.CodeGeneration.CodeGenCommand.Execute(String[] args)'

but if I declare the variable separately:

```
			var valueAsGuid = Guid.Empty;
			if (Guid.TryParse(stringValue, out valueAsGuid)) {

```
there's no exception.

_Copied from original issue: aspnet/Mvc#5869_

--- a/build/dependencies.props
+++ b/build/dependencies.props
@@ -18,7 +18,7 @@
   </PropertyGroup>
   <!-- Partner Versions-->
   <PropertyGroup>
-    <CodeAnalysisVersion>1.3.0</CodeAnalysisVersion>
+    <CodeAnalysisVersion>2.0.0</CodeAnalysisVersion>
     <JsonNetVersion>10.0.1</JsonNetVersion>
     <MicrosoftBuildVersion>15.2.0-preview-000047-02</MicrosoftBuildVersion>
     <NuGetFrameworksVersion>4.0.0-*</NuGetFrameworksVersion>

--- a/test/Shared/MsBuildProjectStrings.cs
+++ b/test/Shared/MsBuildProjectStrings.cs
@@ -184,13 +184,18 @@ namespace WebApplication1
         public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
         {
             app.UseStaticFiles();
-
+            CS7_method(out var i);
             app.UseMvc(routes =>
             {
                 routes.MapRoute(
                     name: ""default"",
                     template: ""{controller=Home}/{action=Index}/{id?}"");
             });
+        }
+
+        private void CS7_method(out int i)
+        {
+            i = 1;
         }
     }
 }

